lib/oplog/tail.go

```diff
@@ -372,11 +366,42 @@ func closeCursor(cursor *mongo.Cursor) {
 // ignored it or failed at some later step.
 func (tailer *Tailer) unmarshalEntry(rawData bson.Raw, denylist *sync.Map, readOrdinal int) (timestamp *primitive.Timestamp, pubs []*redispub.Publication, sendMetricsData func()) {
 	var result rawOplogEntry
+	var ok bool
+	result.Namespace, ok = rawData.Lookup("ns").StringValueOK()
+	if !ok {
+		log.Log.Error("Error unmarshalling oplog namespace entry")
+	}
 
-	err := bson.Unmarshal(rawData, &result)
-	if err != nil {
-		log.Log.Errorw("Error unmarshalling oplog entry", "error", err)
-		return
+	if len(result.Namespace) > 0 { // try to filter early if possible
+		db, _ := parseNamespace(result.Namespace)
+
+		if _, denied := denylist.Load(db); denied {
```

Comment:
The difference is that the second check is post-transaction, the way the original behavior was (though still it only checks the first entry in the transaction).  But maybe we don't care about transactions for filtering?  This might be good to confirm with @alex-goodisman, which originally added the denylist.

Reply:
Actually, I think what you currently have is right. Rereading the parse code, it seems if there’s a command operation, the top level entry’s namespace is just `admin.$cmd`, so we probably do need to check that first sub-entry in the denylist. I’m also realising that we also now check this case (i.e. checking the namespace `admin.$cmd`) in the new denylist check, while we never had to do that before. Should be fine since `admin` shouldn’t be in the denylist, but something to keep in mind.